chacha20/src/rng.rs

```diff
@@ -116,15 +116,16 @@ pub struct StreamId([u32; 3]);
 
 impl From<[u32; 3]> for StreamId {
     fn from(value: [u32; 3]) -> Self {
-        Self(value)
+        Self([value[0].to_le(), value[1].to_le(), value[2].to_le()])
     }
 }
 
 impl From<[u8; 12]> for StreamId {
     fn from(value: [u8; 12]) -> Self {
         let mut result = Self([0u32; 3]);
         for (n, chunk) in result.0.iter_mut().zip(value.chunks_exact(4)) {
-            *n = u32::from_le_bytes(chunk.try_into().unwrap())
+            *n = u32::from_le_bytes(chunk.try_into().unwrap());
+            *n = n.to_le();
```

Comment:
Maybe consolidate this?

```suggestion
            *n = u32::from_le_bytes(chunk.try_into().unwrap()).to_le();
```

Reply:
It is done